File: Frontend/behealthy/app/services/eventHistory.ts
export interface EventHistoryRequest{
    id: string,
    userId: string
    title: string;
    description: string;
    visitTime: Date;
    direction: string;
    doctorName: string;
    doctorPhone: string;
    clinicName: string;
    clinicPhone: string;
    address: string;
}

export const getEventHistory = async (userId : string) => {
    const response = await fetch(`https://localhost:7293/getEventsHistory/${userId}`);

    return response.json();
}

export const createEventHistory = async (medEventRequest : EventHistoryRequest) => {
    await fetch("https://localhost:7293/createEventHistory/", {
        method: "POST",
        headers: {
            "content-type": "application/json",
        },
        body: JSON.stringify(medEventRequest),
    });
};

export const updateEventHistory = async (id : string, medEventRequest : EventHistoryRequest) => {
    await fetch(`https://localhost:7293/updateEventHistory/${id}`, {
        method: "PUT",
        headers: {
            "content-type": "application/json",
        },
        body: JSON.stringify(medEventRequest),
    });
}

export const deleteEventHistory = async (id : string) => {
    await fetch(`https://localhost:7293/deleteEventHistory/${id}`, {
        method: "DELETE",
    });
}

File: Backend/BeHealthy/BeHealthy.Core/Models/EventHistory.cs
namespace BeHealthy.Core.Models;

public class EventHistory
{
    private EventHistory(Guid id, Guid userId, string title, string description,
        DateTime visitTime, string direction, string doctorName,
        string doctorPhone, string clinicName, string clinicPhone, string address)
    {
        Id = id;
        UserId = userId;
        Title = title;
        Description = description;
        VisitTime = visitTime;
        Direction = direction;
        DoctorName = doctorName;
        DoctorPhone = doctorPhone;
        ClinicName = clinicName;
        ClinicPhone = clinicPhone;
        Address = address;
    }

    public Guid Id { get; private set; }

    public Guid UserId { get; private set; }

    public string Title { get; private set; } = string.Empty;

    public string Description { get; private set; } = string.Empty;

    public DateTime VisitTime { get; private set; } = new DateTime();

    public string Direction { get; private set; } = string.Empty;

    public string DoctorName { get; private set; } = string.Empty;

    public string DoctorPhone { get; private set; } = string.Empty;

    public string ClinicName { get; private set; } = string.Empty;

    public string ClinicPhone { get; private set; } = string.Empty;

    public string Address { get; private set; } = string.Empty;

    public static (EventHistory eventHistory, string Error) Create(Guid id, Guid userId, string title, string description,
        DateTime visitTime, string direction, string doctorName,
        string doctorPhone, string clinicName, string clinicPhone, string address)
    {
        var error = string.Empty;

        if (string.IsNullOrEmpty(title) || title.Length > 50)
        {
            error = "Title can not be empty or longer than 50 symbols";
        }
        var eventHistory = new EventHistory(id, userId, title, description, visitTime, direction, doctorName, doctorPhone, clinicName, clinicPhone, address);
        return (eventHistory, error);
    }
}
